### DISCUSSION TITLE
Integrate Blue Carbon estimation code

### DISCUSSION THREAD

**Discussion by NPJuncal** [valybionda/BlueCarbon#8] [General]
Hi @costavale and @MarcioFCMartins,

as I have told you by mail, I have a working code to estimate stocks and fluxes from soil cores data.
Another postdoc from the R packages workshop and I will start a new branch and integrate my code into this project.

I am using linear decompresion, so I wont touch your decompresion function. We will add a function to estimate organic carbon from organic matter (LOI). My stock and flux estimation code usess a modified rectangle method, I have not been able to find the Mathew 2021 reference to see what the trapezoid method is, but for what I have read in your documentation I think the result from my code and the trapezoid should be the same.
I use accumulated organic carbon with depth to extrapolate the stock until one meter, because it worked better that a extrapolation of carbon density.
I will also add a function to estimate the error introduced by the extrapolation: if you have some cores that do reach 1 meter, you can estimate the stock at 1m, and then cut them at 25, 50, 75 and 90 cm and estimate the stock at 1m with extrapolation and compare with the "real" 1 m stock.

For the flux estimations I will assume the user already has a depth-age model (there are libraries that already do this, there is no point in redoing this).

We will probably modifie the data.frame input a bit.

**Comment by costavale** [comment]
Hi @NPJuncal,
as already anticipated by email, I am super happy if we manage to complete the package and provide a fully functional set of functions. Therefore, feel free! 
Regarding the stock estimation, we can later keep the different methods of estimation providing the different equations. Let's check if your results are the same as the previous code and later we can decide.
Regarding the flux estimation, we can then provide a list of the packages doing that in the vignette, with the different functions that can be used.
Let me know if I can provide you with any help. Otherwise, let's wait for the integrations and Happy coding!!

**Comment by NPJuncal** [comment]
Hi @costavale @MarcioFCMartins 
I have uploaded the script with my functions:
transform_om_oc
estimate_h
estimate_stock
test_extrapolation
estimate_flux

Could you try them with your data?
Will work over them a bit more in this two weeks and try to writte some documentation and vignettes.
do you think it would be a good idea to have a meeting to discuss the code and next steps?

**Comment by costavale** [comment]
Hi @NPJuncal,
happy to see that you uploaded your functions to the package!
I will test the functions and let you know.
Let's discuss via mail for the meeting ;)

**Comment by MarcioFCMartins** [comment]
Hi and thanks for the heads-up. I'm hoping to have some time to test your changes early next week, when I do I'll get in touch.
